assets/app/scripts/services/data.js

```diff
@@ -159,6 +159,92 @@ angular.module('openshiftConsole')
     }
   	return deferred.promise;
   };
+  
+// type:      API type (e.g. "pods")
+// object:    API object data(eg. { kind: "Build", parameters: { ... } } )
+// context:   API context (e.g. {project: "..."})
+// opts:      http - options to pass to the inner $http call
+// Returns a promise resolved with response data or rejected with {data:..., status:..., headers:..., config:...} when the delete call completes.
+  DataService.prototype.create = function(type, object, context, opts) {
+  	opts = opts || {};
+  	var deferred = $q.defer();
+    if (context.projectPromise && type !== "projects") {
+      var self = this;
+      context.projectPromise.done(function(project) {
+        $http(angular.extend({
```

Comment:
could refactor this into a function that takes a namespace, then invoke the function with `project.metadata.name` in the case of a `context.projectPromise`, `context.namespace` if so specified, or `""` otherwise